object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,37 @@
+/* exported pick */
+
+/*
+
+***************************** PSEUDOCODE *****************************
+
+- create new object container
+- look through each property in the object
+- compare each property the the values in the array
+- if the value is undefined, move on to the next key
+- if they are equal, add the key value pair to the new object
+- give back the new object
+
+**********************************************************************
+
+*/
+function pick(source, key) {
+  var newObj = {};
+  var earlyEnd = key.length;
+
+  for (var akey in source) {
+    for (var i = 0; i < key.length; i++) {
+      if (source[akey] === undefined) {
+        break;
+      }
+      if (akey === key[i]) {
+        newObj[akey] = source[akey];
+        earlyEnd--;
```

Comment:
I think you can remove the need for this variable if you loop over the array of keys to take instead of the source object as your main loop, and do your `for...in` loop to check if that key exists in the source inside of that loop.